containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceListener.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.openshift.auth.org.hawkular.openshift.namespace;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ *
+ */
+public class NamespaceListener {
+
+    private static final Logger log = Logger.getLogger(NamespaceListener.class);
+
+    // The authentication token used by the filter to connect to the OpenShift Master API
+    private final String token;
+
+    // Maps the namespace name to the namespace id
+    private final Map<String, String> namespaces;
+
+    /**
+     *
+     * @param token The token to be used by the
+     */
+    public NamespaceListener(String token) {
+        log.info("Creating NamespaceListener");
+        this.token = token;
+
+        namespaces = new HashMap<String, String>();
+        log.info("Namespace listener starting");
+        new Thread(new Listener()).start();
```

Comment:
Currently we don't, it get shutdown when the server is shutdown. Adding up a stop method.